tests/ui/consts/const-mut-refs-crate.rs

```diff
@@ -0,0 +1,34 @@
+//@ run-pass
+//@ aux-build:const_mut_refs_crate.rs
+
+#![feature(const_mut_refs)]
+
+//! Regression test for https://github.com/rust-lang/rust/issues/79738
+//! Show how we are duplicationg allocations, even though static items that
+//! copy their value from another static should not also be duplicating
+//! memory behind references.
+
+extern crate const_mut_refs_crate as other;
+
+use other::{
+    inner::{INNER_MOD_BAR, INNER_MOD_FOO},
+    BAR, FOO,
+};
+
+pub static LOCAL_FOO: &'static i32 = &41;
+pub static LOCAL_BAR: &'static i32 = LOCAL_FOO;
+pub static mut COPY_OF_REMOTE_FOO: &'static mut i32 = unsafe { FOO };
```

Comment:
```suggestion
pub static mut COPY_OF_REMOTE_FOO: &'static mut i32 = unsafe { &mut *FOO };
```
Not sure if it makes a difference but that's what the reproducer did.

Reply:
the initializer does not matter, we are already not evaluating the initializer twice anymore.